packages/table-dev-app/src/mutableTable.tsx

```diff
@@ -390,48 +399,109 @@ export class MutableTable extends React.Component<{}, IMutableTableState> {
+        const scrollAbove = top + 0.3 * (bottom - top);
+        const scrollBelow = bottom - 0.3 * (bottom - top);
+        const pos = event.clientY;
+
+        if (pos < scrollAbove && pos > top) {
+            this.setState({ scrollDirection: "UP" });
+            if (this.state.animationRequestId === undefined) {
+                requestAnimationFrame(this.animate);
+            }
+        } else if (pos > scrollBelow && pos < bottom) {
+            this.setState({ scrollDirection: "DOWN" });
+            if (this.state.animationRequestId === undefined) {
+                requestAnimationFrame(this.animate);
+            }
+        } else {
+            this.cancelAnimation();
+        }
+    };
+
     // Renderers
     // =========
 
     private renderTable() {
         return (
-            <Table2
-                bodyContextMenuRenderer={this.renderBodyContextMenu}
-                enableColumnHeader={this.state.enableColumnHeader}
-                enableColumnInteractionBar={this.state.showTableInteractionBar}
-                enableColumnReordering={this.state.enableColumnReordering}
-                enableColumnResizing={this.state.enableColumnResizing}
-                enableFocusedCell={this.state.showFocusCell}
-                enableGhostCells={this.state.showGhostCells}
-                enableMultipleSelection={this.state.enableMultiSelection}
-                enableRowHeader={this.state.enableRowHeader}
-                enableRowReordering={this.state.enableRowReordering}
-                enableRowResizing={this.state.enableRowResizing}
-                getCellClipboardData={this.getCellValue}
-                loadingOptions={this.getEnabledLoadingOptions()}
-                numFrozenColumns={this.state.numFrozenCols}
-                numFrozenRows={this.state.numFrozenRows}
-                numRows={this.state.numRows}
-                onColumnsReordered={this.onColumnsReordered}
-                onColumnWidthChanged={this.onColumnWidthChanged}
-                onCompleteRender={this.onCompleteRender}
-                onCopy={this.onCopy}
-                onFocusedCell={this.onFocus}
-                onRowHeightChanged={this.onRowHeightChanged}
-                onRowsReordered={this.onRowsReordered}
-                onSelection={this.onSelection}
-                onVisibleCellsChange={this.onVisibleCellsChange}
-                ref={this.refHandlers.table}
-                renderMode={this.state.renderMode}
-                rowHeaderCellRenderer={this.renderRowHeader}
-                selectedRegionTransform={this.getSelectedRegionTransform()}
-                selectionModes={this.getEnabledSelectionModes()}
-                selectedRegions={this.state.selectedRegions}
-                styledRegionGroups={this.getStyledRegionGroups()}
-                cellRendererDependencies={[this.state.cellContent]}
+            <div
+                ref={this.refHandlers.tableWrapperRef}
+                style={{
```

Comment:
please write these styles in CSS rather than inline styles. there's are some other classes with custom styling in `table-dev-app/src/index.scss`

Reply:
thanks! moving those now..